CMakeLists.txt

```diff
@@ -2,14 +2,18 @@ cmake_minimum_required(VERSION 3.17)
 
 project(chromo LANGUAGES Fortran)
 
-# Find C/CXX compilers that match the Fortran compiler
-if (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
-  # Compiler version is something like XY.Z.N, we want only XY
-  string(REGEX MATCH "[0-9]+" COMPILER_MAJOR_VERSION ${CMAKE_Fortran_COMPILER_VERSION})
-  find_program(CMAKE_C_COMPILER "gcc-${COMPILER_MAJOR_VERSION}")
-  find_program(CMAKE_CXX_COMPILER "g++-${COMPILER_MAJOR_VERSION}")
-else()
-  message(FATAL_ERROR "Unknown Fortran compiler, cannot search for C/CXX compiler")
+# Force usage of gcc
+# For MacOS turn it off to use Clang
+IF(NOT APPLE)
+  # Find C/CXX compilers that match the Fortran compiler
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
+    # Compiler version is something like XY.Z.N, we want only XY
+    string(REGEX MATCH "[0-9]+" COMPILER_MAJOR_VERSION ${CMAKE_Fortran_COMPILER_VERSION})
+    find_program(CMAKE_C_COMPILER "gcc-${COMPILER_MAJOR_VERSION}")
+    find_program(CMAKE_CXX_COMPILER "g++-${COMPILER_MAJOR_VERSION}")
+  else()
+    message(FATAL_ERROR "Unknown Fortran compiler, cannot search for C/CXX compiler")
+  endif()
```

Comment:
Why use clang on Mac? It is best to use the same compiler for c++ and fortran files. The assumption of this piece is that if you have installed gfortran, you have installed the whole gcc suite, so that the c++ compiler is also available.

Unless there is a specific reason to prefer clang on Mac, please do not expections to a general rule.

Reply:
Yes, there is a reason. The build fails for MacOS arm64 if gcc is enforced. It works only with Clang. On MacOS native it still works.

Reply:
Ok, can you then please restrict the exception to APPLE + arch64?

Reply:
I will try to use `CMAKE_OSX_ARCHITECTURES EQUAL "arm64"` to restrict it to arm64

Reply:
`CMAKE_OSX_ARCHITECTURES EQUAL "arm64"` seems doesn't work. @HDembinski, do you know how to get an architecture in CMake?

Reply:
Are you sure it is arm64 and not arch64?

Reply:
It is arm64, but the problem was with EQUAL. It should be STREQUAL.